### PR TITLE
✨ feat (CHASE Frontend): Font Awesome Pro

### PR DESCRIPTION
**Pull request by TadeSF**
branch: font-awesome-pro

This PR includes the Font Awesome Pro Icon Library. It also makes some changes to individual icons

### CODE REVIEW

**Comment by TadeSF** [comment]
Just realised we have a CI/CD Problem with FontAwesome now since we need to set the token for the npm package. @m1212e I can give you the token without a problem, but could you handle the CI/CD Setup? FontAwesome Docs with the relevant information can be found [here](https://fontawesome.com/docs/web/setup/packages#_1-configure-access)

**Comment by m1212e** [comment]
For the gh actions you can add a secret env var value here: https://github.com/DeutscheModelUnitedNations/munify/settings/secrets/actions/new

Not sure if you are allowed to do so, but you have admin access to the repo so you should be, I guess. The key must be FONTAWESOME_NPM_AUTH_TOKEN

**Comment by m1212e** [comment]
The added file tells npm to access a private registry at a specific domain with the access key from the env vars. As stated here: https://fontawesome.com/docs/web/setup/packages#alternate-per-project-setup-using-environment-variables

**Comment by TadeSF** [comment]
Thanks! But I am not sure this works now. I added the Key like you said, but GH Actions doesnt seem to access the env variable...

**Comment by m1212e** [comment]
My bad, forgot to set the secret value as env var for the action

**Comment by m1212e** [comment]
Since we now need to add credentials for the protected repo we should include docs for this in the readme, so new people know why their install fails

**Comment by m1212e** [comment]
Mhh the install works, but it cant find the modules imported. Does this work on your local machine?

**Comment by TadeSF** [comment]
I've configured FontAwesome globally on my machine, that works like a breeze. But you can try it locally if you want, I'll send you the token another way

**Comment by m1212e** [comment]
Ok, I got it on my machine now and I get the same issue with some packages not found:

```
npm run build

> chase-frontend@0.1.0 build
> next build

- warn Found lockfile missing swc dependencies, patching...
- warn Lockfile was successfully patched, please run "npm install" to ensure @next/swc dependencies are downloaded
Failed to compile.

./app/chair/attendees/page.tsx
Module not found: Can't resolve '@fortawesome/pro-solid-svg-icons'

https://nextjs.org/docs/messages/module-not-found

./app/chair/layout.tsx
Module not found: Can't resolve '@fortawesome/pro-solid-svg-icons'

https://nextjs.org/docs/messages/module-not-found

./app/chair/voting/page.tsx
Module not found: Can't resolve '@fortawesome/pro-solid-svg-icons'

https://nextjs.org/docs/messages/module-not-found

./app/chair/whiteboard/page.tsx
Module not found: Can't resolve '@fortawesome/pro-solid-svg-icons'

https://nextjs.org/docs/messages/module-not-found

./app/participant/layout.tsx
Module not found: Can't resolve '@fortawesome/pro-solid-svg-icons'

https://nextjs.org/docs/messages/module-not-found


> Build failed because of webpack errors
```

which makes sense, since those packages are not in the package.json -> not installed. Is this right? Or am I missing something?

**Comment by TadeSF** [comment]
Thanks, that explains alot! xD Sorry that I missed that.
Now I have just a few type errors to get through, I'll be back.

**Comment by TadeSF** [comment]
Okay, finally the build works. But now the linter complains again because of missing credentials. I know this is really annoying, but could you please look into the problem again?

**Comment by TadeSF** [comment]
I finally got it. Yay. And its official: this was stupid and I am to blame... Well, some things you need to learn the hard way I guess...

But: This PR is then finally ready. It is a little bit hacky, since there was an annoying type error forcing me to go through every file and adding a `as IconProp` to the FontAwesomeIcon Components since the types weirdly did not add up. But this will do for now.

File: chase/frontend/.npmrc
Comment (m1212e) [inline chase/frontend/.npmrc:1]: After the switch to bun we will need a bunfig.toml

**Review by m1212e — APPROVED** [review]
Very nice!

**Pull request merged**